main.py

```diff
@@ -0,0 +1,90 @@
+import argparse
+from datetime import datetime
+
+import WeatherManComputations
+
+OKBLUE = '\033[34m'
+OKRED = '\033[31m'
+Default = "\033[39m"
+
+
+def problem_1(year_data):
+    max_temp_data = max(year_data, key=lambda x: int(x.max_temp) if x.max_temp else -200)
+    min_temp_data = min(year_data, key=lambda x: int(x.min_temp) if x.min_temp else 200)
+    max_humid = max(year_data, key=lambda x: int(x.max_humidity) if x.max_humidity else 0)
+    print("Highest: ",
+          max_temp_data.max_temp + " C on " + datetime.strptime(max_temp_data.pkt, '%Y-%m-%d').strftime("%B %d"))
+    print("Lowest: ",
+          min_temp_data.min_temp + " C on " + datetime.strptime(min_temp_data.pkt, '%Y-%m-%d').strftime("%B %d"))
+    print("Humid: ", max_humid.max_humidity + " % on " + datetime.strptime(max_humid.pkt, '%Y-%m-%d').strftime("%B %d"))
+
+
+def problem_2(year_data):
+    max_avg_temp = max(year_data, key=lambda x: int(x.mean_temp) if x.mean_temp else -200)
+    min_avg_temp = min(year_data, key=lambda x: int(x.mean_temp) if x.mean_temp else 200)
+    print("Highest Average: " + max_avg_temp.mean_temp + " C")
+    print("Min Average: " + min_avg_temp.mean_temp + " C")
+    print("Average Mean Humidity: " + str(average_humidity(year_data)) + " %")
+
+
+def average_humidity(year_data):
+    sum_humidity = 0
+    for data in year_data:
+        if data.mean_humidity:
+            sum_humidity = sum_humidity + int(data.mean_humidity)
+
+    return sum_humidity / len(year_data)
+
+
+def problem_3(year_data):
+    print(datetime.strptime(year_data[0].pkt, '%Y-%m-%d').strftime('%B %Y'))
+    for data in year_data:
+        if data.max_temp or data.min_temp:
+            print(
+                f"{datetime.strptime(data.pkt, '%Y-%m-%d').strftime('%d')} {OKRED}{get_plus(int(data.max_temp))} {Default} {data.max_temp} C")
+            print(
+                f"{datetime.strptime(data.pkt, '%Y-%m-%d').strftime('%d')} {OKBLUE}{get_plus(int(data.min_temp))} {Default} {data.min_temp} C")
+
+
+def bonus_task(year_data):
+    print(datetime.strptime(year_data[0].pkt, '%Y-%m-%d').strftime('%B %Y'))
+    for data in year_data:
+        if data.max_temp or data.min_temp:
+            print(
+                f"{datetime.strptime(data.pkt, '%Y-%m-%d').strftime('%d')} {OKRED}{get_plus(int(data.max_temp))}{OKBLUE}{get_plus(int(data.min_temp))} {Default} {data.max_temp} -{data.min_temp}C ")
+
+
+def get_plus(count):
+    plus_str = ""
+    for i in range(count):
+        plus_str += "+"
+    return plus_str
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("dir", help="Pass an year value")
+    parser.add_argument("-e", help="Pass an year value")
+    parser.add_argument("-a", help="Pass an year value")
+    parser.add_argument("-c", help="Pass an year value")
+    parser.add_argument("-b", help="Pass an year value")
+    args = parser.parse_args()
+    computations = WeatherManComputations.Computations(args.dir + "/*.txt")
```

Comment:
Let's determine the year and month passed here. and pass that to the Weather Data Loader Class constructor to use and pick up those files.
And it a option only want's year then we can wildcard the month with '*' constant as keyword argument in Weather Data Loader Class constructor.